packages/react/src/auto/polaris/PolarisAutoTable.tsx

```diff
@@ -46,15 +59,48 @@ export const PolarisAutoTable = <
 ) => {
   const { onClick } = props;
 
-  const [{ rows, columns, metadata, fetching, error, page, search }, refresh] = useTable<GivenOptions, SchemaT, FinderFunction, Options>(
-    props.model,
-    {
-      select: props.select,
-      columns: props.columns,
-      pageSize: props.pageSize,
-      live: props.live,
-    } as any
+  const [{ rows, columns, metadata, fetching, error, page, search, sort }, refresh] = useTable<
+    GivenOptions,
+    SchemaT,
+    FinderFunction,
+    Options
+  >(props.model, {
+    select: props.select,
+    columns: props.columns,
+    pageSize: props.pageSize,
+    live: props.live,
+  } as any);
+
+  const [sortColumnIndex, setSortColumnIndex] = useState<number | undefined>(
+    columns ? getColumnIndex(columns, props.sort ? Object.keys(props.sort)[0] : undefined) : undefined
+  );
+  const [setColumnIndexToDefault, setSetColumnIndexToDefault] = useState(true);
```

Comment:
This variable is really confusing to the people who read code that `setColumnIndexToDefault` is actually a state value, not the function that modifies the value. Maybe `defaultColumnIndex` and `setDefaultColumnIndex`?

Also, some states seem to be for triggering the `useEffect` and updating another state. Is it possible to instead use hooks like `useMemo` or `useCallback` to reduce the number of states needed?

Reply:
The `useEffect` is so that when `column` loads, the filter icon updates, previously, it was not always pointing to the correct index because column does some time to load.

From what I understand, `useMemo` or `useCallback` don't update when `column` updates. 

I limited the dependancy array so useEffect is only triggered when `columns` updates.